### PR TITLE
Make GH evolve GaugeWave instead of KerrSchild

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

This PR is intended for code sharing, not to be merged. It evolves a gauge wave instead of KerrSchild with the GeneralizedHarmonic executable.

Note: to actually get an accurate evolution, increase resolution and use a better time stepper in the `GaugeWave.yaml` input file. The example here uses simple settings so that the automated test doesn't time out.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by kidder** [comment]
@wthrowe @geoffrey4444 can we close this?

**Comment by geoffrey4444** [comment]
I made this for code sharing, hence it is marked draft. I'm fine if you close it, though @wthrowe and I might still find it helpful for the local time stepping exploration we are pursuing.

**Comment by wthrowe** [comment]
@kidder I would like to leave this open for discussion until I have a working gauge wave simulation.

@geoffrey4444 This input file is suitable for a code test, but clearly not for a real run.  (It has one point per wavelength.)  I'm having trouble setting up a working 3D simulation.  Do you have a setup that works for that?

File: tests/InputFiles/GeneralizedHarmonic/GaugeWave.yaml [hunk 1]
```diff
@@ -16,21 +16,17 @@ Evolution:
...
-      UseEquiangularMap: true
-      AspectRatio: 1.0
-      UseLogarithmicMap: true
-      RadialBlockLayers: 1
+  Brick:
+    LowerBound: [-2.0, -2.0, -2.0]
+    UpperBound: [2.0, 2.0, 2.0]
+    InitialRefinement: [0, 0, 0]
+    InitialGridPoints: [5, 5, 5]
+    IsPeriodicIn: [true, true, true]
```
Comment (geoffrey4444) [inline tests/InputFiles/GeneralizedHarmonic/GaugeWave.yaml:25]: In the last long, higher-resolution run I ran, I used
~~~

DomainCreator:
    Brick:
      LowerBound: [-0.5, -0.5, -0.5]
      UpperBound: [0.5, 0.5, 0.5]
      InitialRefinement: [0, 0, 0]
      InitialGridPoints: [5, 5, 5]
      IsPeriodicIn: [false, false, false]
AnalyticSolution:
  GaugeWave:
    Amplitude: 0.01
    Wavelength: 10.0
~~~

**Comment by geoffrey4444** [comment]
@wthrowe please send me your public key and I'll give you an account, so you can see my spec and spectre runs of the gauge wave:

`ocean:/home/geoffrey/spectre_tests/GaugeWave/spec_vs_spectre`

**Comment by wthrowe** [comment]
I got a 3D run working.  Turns out the problem didn't really have much to do with the domain or solution or anything, it was just a standard CFL instability.  Since that's solved, I think we can close this.

I'll send out a report once I've made some basic measurements.

**Comment by geoffrey4444** [comment]
Let’s see if I can get #1611 merged, which prints the minimum grid spacing so you have an idea what cfl for your domain is. A student was on this but then graduated, and the PR languished.

**Pull request closed without merging**